sql/core/src/test/scala/org/apache/spark/sql/analysis/resolver/NameScopeSuite.scala

```diff
@@ -103,263 +104,113 @@ class NameScopeSuite extends PlanTest with SQLConfHelper {
 
     assert(
-      nameScope.matchMultipartName(Seq("cOL1")) == NameTarget(
-        candidates = Seq(col1Integer.withName("cOL1")),
-        allAttributes = Seq(col1Integer, col2Integer, col3Boolean, col4String)
+      nameScope.output == Seq(
+        col1Integer,
+        col3Boolean,
+        col2Integer,
+        col2Integer,
+        col3BooleanOther,
+        col4String
       )
     )
-    assert(
-      nameScope.matchMultipartName(Seq("CoL2")) == NameTarget(
-        candidates = Seq(col2Integer.withName("CoL2")),
-        allAttributes = Seq(col1Integer, col2Integer, col3Boolean, col4String)
-      )
+
+    checkOnePartNameLookup(
+      nameScope,
+      name = "cOL1",
+      candidates = Seq(col1Integer)
     )
-    assert(
-      nameScope.matchMultipartName(Seq("col3")) == NameTarget(
-        candidates = Seq(col3Boolean.withName("col3")),
-        allAttributes = Seq(col1Integer, col2Integer, col3Boolean, col4String)
-      )
+    checkOnePartNameLookup(
+      nameScope,
+      name = "CoL2",
+      candidates = Seq(col2Integer, col2Integer)
     )
-    assert(
-      nameScope.matchMultipartName(Seq("COL4")) == NameTarget(
-        candidates = Seq(col4String.withName("COL4")),
-        allAttributes = Seq(col1Integer, col2Integer, col3Boolean, col4String)
-      )
+    checkOnePartNameLookup(
+      nameScope,
+      name = "col3",
+      candidates = Seq(col3Boolean, col3BooleanOther)
     )
-    assert(
-      nameScope.matchMultipartName(Seq("col5")) == NameTarget(
-        candidates = Seq.empty,
-        allAttributes = Seq(col1Integer, col2Integer, col3Boolean, col4String)
-      )
+    checkOnePartNameLookup(
+      nameScope,
+      name = "COL4",
+      candidates = Seq(col4String)
     )
-  }
-
-  test("Duplicate attribute names from one plan") {
-    val nameScope = new NameScope
-
-    nameScope("table1") = Seq(col1Integer, col1Integer)
-    nameScope("table1") = Seq(col1IntegerOther)
-
-    assert(nameScope.getAllAttributes == Seq(col1Integer, col1Integer, col1IntegerOther))
-
-    nameScope.matchMultipartName(Seq("col1")) == NameTarget(
-      candidates = Seq(col1Integer, col1IntegerOther)
-    )
-  }
-
-  test("Duplicate attribute names from several plans") {
-    val nameScope = new NameScope
-
-    nameScope("table1") = Seq(col1Integer, col1IntegerOther)
-    nameScope("table2") = Seq(col1Integer, col1IntegerOther)
-
-    assert(
-      nameScope.getAllAttributes == Seq(
-        col1Integer,
-        col1IntegerOther,
-        col1Integer,
-        col1IntegerOther
-      )
-    )
-
-    nameScope.matchMultipartName(Seq("col1")) == NameTarget(
-      candidates = Seq(
-        col1Integer,
-        col1IntegerOther,
-        col1Integer,
-        col1IntegerOther
-      )
+    checkOnePartNameLookup(
+      nameScope,
+      name = "col5",
+      candidates = Seq.empty
     )
   }
 
   test("Expand star") {
-    val nameScope = new NameScope
-
-    nameScope("table") =
+    var nameScope = new NameScope(
```

Comment:
Do you really need the var?
```suggestion
    val nameScope = new NameScope(
```